### PR TITLE
[MWPW-140096] Genfill Marquee images are blinking during auto play and on click as well in Firefox browser

### PR DESCRIPTION
**Pull request by drashti1712**
* Bug fix - Image flickering issue on firefox browser (Genfill)

Resolves: [MWPW-140096](https://jira.corp.adobe.com/browse/MWPW-140096)

**Test URLs:**
- Before: https://main--cc--adobecom.hlx.live/drafts/drashti/MWPW-137345/genfill-container?martech=off
- After: https://genfill-flicker--cc--drashti1712.hlx.live/drafts/drashti/MWPW-137345/genfill-container?martech=off

### CODE REVIEW

**Review by aishwaryamathuria — APPROVED** [review]

File: creativecloud/features/genfill/genfill-interactive.js [hunk 1]
```diff
@@ -1,75 +1,74 @@
...
+    mDiv.insertBefore(entcmtEl.cloneNode(true), mDiv.firstElementChild);
+  });
+}
+
+function handleClick(a, viewport, deviceConfig) {
+  const img = a.querySelector('img');
+  const currIndex = deviceConfig[viewport].index;
+  const nextIndex = (currIndex + 1) % deviceConfig[viewport].srcList.length;
+  img.src = deviceConfig[viewport].srcList[nextIndex];
+  img.alt = deviceConfig[viewport].altList[nextIndex];
```
Comment (aishwaryamathuria) [inline creativecloud/features/genfill/genfill-interactive.js:21]: Image alt ahould be same as the image and not the daa-ll

File: creativecloud/features/genfill/genfill-interactive.js [hunk 2]
```diff
@@ -1,75 +1,74 @@
...
-    a.addEventListener('click', () => {
-      clickConfig.isImageClicked = true;
-      if (clickConfig.autocycleInterval) clearInterval(clickConfig.autocycleInterval);
-      handleTransition(aTags, i);
-    });
+function processMedia(ic, miloUtil, autoCycleConfig, deviceConfig, viewport) {
+  const media = miloUtil.createTag('div', { class: `media ${viewport}-only` });
+  const a = miloUtil.createTag('a', { class: 'genfill-link' });
+  const img = miloUtil.createTag('img', { class: 'genfill-image' });
+  [img.alt] = [deviceConfig[viewport].altList];
```
Comment (aishwaryamathuria) [inline creativecloud/features/genfill/genfill-interactive.js:42]: Image alt ahould be same as the image and not the daa-ll

**Review by aishwaryamathuria — CHANGES_REQUESTED** [review]

File: creativecloud/features/genfill/genfill-interactive.js [hunk 3]
```diff
@@ -1,75 +1,82 @@
...
+    : `Image-${v}|Marquee|${hText}`;
+  return altTxt;
+}
+
+function handleClick(a, v, deviceConfig, hText) {
+  const img = a.querySelector('img');
+  const currIndex = deviceConfig[v].index;
+  const nextIndex = (currIndex + 1) % deviceConfig[v].srcList.length;
+  img.src = deviceConfig[v].srcList[nextIndex];
+  const alt = deviceConfig[v].altList[nextIndex];
```
Comment (aishwaryamathuria) [inline creativecloud/features/genfill/genfill-interactive.js:28]:
```suggestion
  const alt = deviceConfig[v].altList[nextIndex];
  if (alt) img.alt = alt
```

Comment (aishwaryamathuria) [inline creativecloud/features/genfill/genfill-interactive.js:28]:
```suggestion
  const alt = deviceConfig[v].altList[nextIndex];
  if (alt) img.alt = alt;
```

File: creativecloud/features/genfill/genfill-interactive.js [hunk 4]
```diff
@@ -1,75 +1,82 @@
...
-      clickConfig.isImageClicked = true;
-      if (clickConfig.autocycleInterval) clearInterval(clickConfig.autocycleInterval);
-      handleTransition(aTags, i);
-    });
+function processMedia(ic, miloUtil, autoCycleConfig, deviceConfig, v, hText) {
+  const media = miloUtil.createTag('div', { class: `media ${v}-only` });
+  const a = miloUtil.createTag('a', { class: 'genfill-link' });
+  const img = miloUtil.createTag('img', { class: 'genfill-image' });
+  const alt = [deviceConfig[v].altList];
+  [img.alt] = generateDaaLL(hText, alt, v);
```
Comment (aishwaryamathuria) [inline creativecloud/features/genfill/genfill-interactive.js:50]: Alt should not be modified value.. daa-ll should be in a tag only.
if (alt) img.alt = alt

Comment (aishwaryamathuria) [inline creativecloud/features/genfill/genfill-interactive.js:50]:
```suggestion
  if (alt) img.alt = alt;
```

**Review by aishwaryamathuria — APPROVED** [review]

**Comment by Anupamasonnalagi** [comment]
Verified the issue. It's fine now.
https://genfill-flicker--cc--drashti1712.hlx.page/creativecloud/animation/discover/testdraftsfirefly/genfill/psgenfill

**Review by salonijain3 — APPROVED** [review]

**Pull request merged**